packages/wmr/src/wmr-middleware.js

```diff
@@ -316,105 +316,126 @@ export const TRANSFORMS = {
+						}
+						return '/@' + prefix + '/' + spec;
+					});
+
+					// foo.css --> foo.css.js (import of CSS Modules proxy module)
+					if (spec.match(/\.(css|s[ac]ss)$/)) spec += '.js';
+
+					// Bare specifiers are npm packages:
+					if (!/^\0?\.?\.?[/\\]/.test(spec)) {
+						const meta = normalizeSpecifier(spec);
+
+						// // Option 1: resolve all package verions (note: adds non-trivial delay to imports)
+						// await resolvePackageVersion(meta);
+						// // Option 2: omit package versions that resolve to the root
+						// // if ((await resolvePackageVersion({ module: meta.module, version: '' })).version === meta.version) {
+						// // 	meta.version = '';
+						// // }
+						// spec = `/@npm/${meta.module}${meta.version ? '@' + meta.version : ''}${meta.path ? '/' + meta.path : ''}`;
+
+						// Option 3: omit root package versions
+						spec = `/@npm/${meta.module}${meta.path ? '/' + meta.path : ''}`;
 					}
-					return '/@' + prefix + '/' + spec;
-				});
-
-				// foo.css --> foo.css.js (import of CSS Modules proxy module)
-				if (spec.match(/\.(css|s[ac]ss)$/)) spec += '.js';
 
-				// Bare specifiers are npm packages:
-				if (!/^\0?\.?\.?[/\\]/.test(spec)) {
-					const meta = normalizeSpecifier(spec);
-
-					// // Option 1: resolve all package verions (note: adds non-trivial delay to imports)
-					// await resolvePackageVersion(meta);
-					// // Option 2: omit package versions that resolve to the root
-					// // if ((await resolvePackageVersion({ module: meta.module, version: '' })).version === meta.version) {
-					// // 	meta.version = '';
-					// // }
-					// spec = `/@npm/${meta.module}${meta.version ? '@' + meta.version : ''}${meta.path ? '/' + meta.path : ''}`;
+					const modSpec = spec.replace('../', '/').replace('./', '/');
+					mod.dependencies.add(modSpec);
+					if (!moduleGraph.has(modSpec)) {
+						moduleGraph.set(modSpec, { dependencies: new Set(), dependents: new Set(), acceptingUpdates: false });
+					}
 
-					// Option 3: omit root package versions
-					spec = `/@npm/${meta.module}${meta.path ? '/' + meta.path : ''}`;
-				}
+					const specModule = moduleGraph.get(modSpec);
+					specModule.dependents.add(importer);
+					if (specModule.stale) {
+						return spec + `?t=${Date.now()}`;
+					}
 
-				const modSpec = spec.replace('../', '/').replace('./', '/');
-				mod.dependencies.add(modSpec);
-				if (!moduleGraph.has(modSpec)) {
-					moduleGraph.set(modSpec, { dependencies: new Set(), dependents: new Set(), acceptingUpdates: false });
+					return spec;
 				}
+			});
 
-				const specModule = moduleGraph.get(modSpec);
-				specModule.dependents.add(importer);
-				if (specModule.stale) {
-					return spec + `?t=${Date.now()}`;
-				}
+			writeCacheFile(out, id, code);
 
-				return spec;
+			return code;
+		} catch (e) {
+			if (code && e.loc && e.loc.line) {
+				const splittedCode = code.split('\n');
+
+				e.codeFragment = `
```

Comment:
I wonder if we could reuse the code frame code from here:

https://github.com/preactjs/wmr/blob/57afc7d8395d1bdea37b8ad40fc365b03be96267/packages/wmr/src/lib/output-utils.js#L64

Should be as easy as calling that function and wrapping it in a `kl.stripColors()` which removes the ASCII color notation codes from the string

Reply:
We should probably have a consistent terminology throughout our code base: "code frame" vs "code fragment"